HCP Lambda + EKS Tutorial Infrastructure

webdog opened the pull request:
Ready for review

im2nguyen reviewed (commented):
can you move everything in the `terraform` directory into the root `datacenter-deploy-hcp-eks-lambda`? it doesn't make sense for us to add another layer of directories when it's not required

for now, we're adding the eks-client module in this project as a stop-gap. when we integrate the changes into the main `eks-client` module. i would like us to move the sub-module and pull it from terraform registry

a nit -- the number of yaml files in `eks-client/service-defaults` is low-key ridiculous. we should put them into one file, delimited by `---`. this will make it easier to maintain lol

im2nguyen commented:
also, please include the payments-lambda file source

webdog commented:
>a nit -- the number of yaml files in eks-client/service-defaults is low-key ridiculous. we should put them into one file, delimited by ---. this will make it easier to maintain lol

Low-key I agree. Unfortunately, it's a side effect of the kubectl provider. It doesn't support multiple yaml statements delimited by dashes. I tried; didn't work.

webdog commented:
* Fixed a region issue found by @eddie-rowe 
* Added lambda payments source code
* moved everything from ../terraform up one level.

cc @im2nguyen

im2nguyen commented on datacenter-deploy-hcp-eks-lambda/lambda-payments/main.go:
```diff
@@ -0,0 +1,70 @@
+		Message:        "Payment processed successfully, card details returned for demo purposes, not for production",
+		ID:             uuid.New().String(),
+		CardPlaintext:  requestBody.Number,
+		CardCiphertext: "Encryption Enabled",
```
Please re-add Lambda to this

webdog commented on datacenter-deploy-hcp-eks-lambda/lambda-payments/main.go:
```diff
@@ -0,0 +1,70 @@
+		Message:        "Payment processed successfully, card details returned for demo purposes, not for production",
+		ID:             uuid.New().String(),
+		CardPlaintext:  requestBody.Number,
+		CardCiphertext: "Encryption Enabled",
```
```suggestion
CardCiphertext: "Encryption Enabled (Lambda)",
```

webdog replied:
@im2nguyen ^^
